From: Julian Ospald <hasufell@gentoo.org>
Date: Thu Nov 15 23:59:49 UTC 2012
Subject: build system

respect CFLAGS, use rdynamic
don't use AM_CFLAGS in configure.in
fix macros for automake-1.13

--- configure.in
+++ configure.in
@@ -1,16 +1,13 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT(configure.in)
-AM_INIT_AUTOMAKE(pnmixer, 0.6-dev)
-AM_CONFIG_HEADER(config.h)
+AC_INIT([pnmixer], [0.6-dev])
+AM_INIT_AUTOMAKE
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
-OLD_CFLAGS=$CFLAGS
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
-CFLAGS=$OLD_CXXFLAGS
 
 pkg_modules="gtk+-2.0 >= 2.16.0"
 
@@ -37,12 +34,13 @@
 
 if test x"$debugit" = x"yes"; then
     AC_DEFINE([DEBUG],[],[Debug Mode])
-    AM_CFLAGS="$AM_CFLAGS -g -Wall -Werror -Wno-uninitialized -Wformat -Wformat-security -O0 -export-dynamic"
 else
     AC_DEFINE([NDEBUG],[],[No-debug Mode])
-    AM_CFLAGS="$AM_CFLAGS -O2 -export-dynamic"
 fi
 
+CFLAGS="$CFLAGS -Wall -Wno-uninitialized -Wformat -Wformat-security"
+LDFLAGS="-rdynamic $LDFLAGS"
+
 ##########################################################################
 
 # Make sure we have X11, and also have it added to LIBS
@@ -74,7 +72,6 @@
    AC_DEFINE([HAVE_LIBN],1,[Defined if you have libnotify])
 fi
 
-AC_SUBST([AM_CFLAGS])
 
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 AC_SUBST(PACKAGE_CFLAGS)
--- src/Makefile.am
+++ src/Makefile.am
@@ -1,6 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = @AM_CFLAGS@
 
 INCLUDES = \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
